### PR TITLE
Add support for offset correction

### PR DESCRIPTION
**Pull request by twam**

### CODE REVIEW

**Comment by nickoe** [comment]
@matthewlai Will this offset correction be added? I also find that the 3D viewer on JLCPCB do show my IDC-Header's being a bit offset after a rotation fix of 270 degrees.

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py [hunk 1]
```diff
@@ -31,6 +33,13 @@
 _LOGGER = Log()
 
 
+@dataclass
+class DatabaseEntry:
+    rotation: int
+    offset_x: Optional[float] = None
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:39]: I would prefer just setting 0 offset on parsing if they don't exist, for consistency with rotation

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py [hunk 2]
```diff
@@ -108,20 +131,23 @@ def FixRotations(input_filename, output_filename, db):
...
-
-                if last_correction is not None:
+                        last_entry = entry
+                        last_pattern = pattern
+
+                del entry
+                del pattern
+
+                if last_entry is not None:
+                    _LOGGER.logger.info(f"Footprint {row[package_index]} matched {last_pattern.pattern}. Applying {last_entry.rotation} deg rotation correction")
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:149]: Log at the end of the if block instead and include both rotation and offset?

**Review by matthewlai — CHANGES_REQUESTED** [review]
Thanks for the PR. Sorry for the delay - life got in the way!

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py (hunk 1 above)
Comment (twam) [reply jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py]: Sure.

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py (hunk 2 above)
Comment (twam) [reply jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py]: Sure.

**Comment by matthewlai** [comment]
Amazing. Thanks for the contribution and for your patience!

**Pull request merged**